pkg/controller/logstorage/users/users_controller.go

```diff
@@ -418,23 +467,37 @@ func (r *UsersCleanupController) cleanupStaleUsers(ctx context.Context, logger l
 
 		lu := utils.LinseedUser(clusterID, t.Spec.ID)
 		dashboardsUser := utils.DashboardUser(clusterID, t.Spec.ID)
+		kibanaUser := utils.KibanaUser(clusterID, t.Spec.ID)
+		deletedUsers := 0
 		for _, user := range allESUsers {
-			if user.Username == lu.Username || user.Username == dashboardsUser.Username {
-				err = esClient.DeleteUser(ctx, &user)
+			switch user.Username {
+			case lu.Username, dashboardsUser.Username:
+				err = esClient.DeleteUser(ctx, &user, false)
+				if err != nil {
+					logger.Error(err, "Failed to delete elastic user")
+				} else {
+					deletedUsers++
+				}
+			case kibanaUser.Username:
+				err = esClient.DeleteUser(ctx, &user, true)
 				if err != nil {
 					logger.Error(err, "Failed to delete elastic user")
+				} else {
+					deletedUsers++
 				}
+			}
+		}
 
-				// Remove the finalizer from the tenant to allow it to be deleted.
-				if stringsutil.StringInSlice(userCleanupFinalizer, t.GetFinalizers()) {
-					t.SetFinalizers(stringsutil.RemoveStringInSlice(userCleanupFinalizer, t.GetFinalizers()))
-					if err = r.client.Update(ctx, &t); err != nil {
-						logger.Error(err, "Failed to remove user cleanup finalizer from tenant")
-					}
+		if deletedUsers > 0 {
```

Comment:
I guess this is maintaining the same behavior as before but it seems like we should only remove the finalizer if all 3 users have been removed. I think if removal of any of them failed then the finalizer should remain. WDYT?